src/components/features/Comments.jsx

```diff
@@ -25,29 +28,63 @@ const Comments = ({ articleId }) => {
       <p className="error-message">Something went wrong - please try again</p>
     );
 
+  console.log(comments);
+
   return (
     <section className="comments container">
-      <ol className="comments-ol">
-        {comments.map((comment) => {
-          return (
-            <li key={comment.comment_id} className="comment-card">
-              <h4 className="commenter-name">@{comment.author}</h4>
-              <p className="comment-posted">
-                {new Date(comment.created_at).toLocaleDateString({
-                  weekday: "long",
-                  year: "numeric",
-                  month: "long",
-                  day: "numeric",
-                })}
-              </p>
-              <p className="comment-body">{comment.body}</p>
-              <p className="comment-votes">Votes: {comment.votes}</p>
-              <button className="comment-upvote">👍</button>
-              <button className="comment-downvote">👎</button>
-            </li>
-          );
-        })}
-      </ol>
+      <section className="post-comment">
+        <form className="comment-form">
+          <textarea
+            value={newComment}
+            placeholder="Add a comment..."
+            type="text"
+            className="comment-input"
+            onChange={(e) => {
+              setNewComment(e.target.value);
+            }}
+          />
+          <button
+            type="submit"
+            className="comment-button"
+            onClick={(e) => {
+              e.preventDefault();
+              postComment(articleId, user.username, newComment);
+              optRenderComment(
+                comments,
+                newComment,
+                user.username,
+                setComments
+              );
+              setNewComment("");
```

Comment:
At the moment if we post a comment offline, it optimistically renders which is fine, then disappears when we refresh the page. I think this functionality is good, but could we get some sort of message to the user to let them know their post has failed?

Reply:
Cheers John, will take a look!